lib/redis/store/factory.rb

```diff
@@ -84,6 +84,7 @@ def self.extract_host_options_from_uri(uri)
 
           options[:db]        = db.to_i   if db
           options[:namespace] = namespace if namespace
+          options[:username]  = uri.user if uri.user
```

Comment:
Both Ruby [1] and Go [2] Redis clients support URLs in the form:

```
unix://<user>:<password>@</path/to/redis.sock>?db=<db_number>
```

* [1] redis-client v0.20.0 added support: https://github.com/redis-rb/redis-client/pull/170
* [2] https://github.com/redis/go-redis/blob/90c7a414ac819ab9e15cc01cd0bfa6f357236585/options.go#L247